File: 섹션06. DFS - 경우의 수 만들기/07-01. BOJ 6603 - 로또.cpp
#include <stdio.h>

#define MAX (13 + 3)

int N;

int num_of_cases[MAX];
int number[MAX];

void input()
{
	scanf("%d", &N);

	for (int i = 1; i <= N; i++) scanf("%d", &number[i]);
}

void printCases()
{
	for (int i = 0; i < 6; i++) printf("%d ", num_of_cases[i]);
	putchar('\n');
}

void DFS(int depth, int start)
{
	if (depth == 6)
	{
		printCases();
		return;
	}

	for (int i = start; i <= N; i++)
	{
		num_of_cases[depth] = number[i];

		DFS(depth + 1, i + 1);
	}
}

int main()
{
	while (1)
	{
		input();

		if (N == 0) break;

		DFS(0, 1);

		putchar('\n');
	}

	return 0;
}

File: Python/섹션06. DFS - 경우의 수 만들기/07-01. BOJ 6603 - 로또.py
MAX = 13 + 3

num_of_cases = [0] * MAX
number = [0] * MAX

def input_data():
    global N
    
    line = list(map(int, input().split()))
    N = line[0]
    number[1:N + 1] = line[1:]
    
def print_cases():
    print(*num_of_cases[:6])

def DFS(depth, start):
    if depth == 6:
        print_cases()
        return

    for i in range(start, N + 1):
        num_of_cases[depth] = number[i]
        
        DFS(depth + 1, i + 1)

while True:
    input_data()
    
    if N == 0: break
    
    DFS(0, 1)
    
    print("")
    